app/models/audience_segment.rb

```diff
@@ -0,0 +1,59 @@
+class AudienceSegment < ApplicationRecord
+  # enum does not like names that start with "not_"
+  enum type_of: {
+    testing: 0, # never matches anyone, used in test factory
+    trusted: 1,
+    posted: 2,
+    no_posts_yet: 3,
+    dark_theme: 4,
+    light_theme: 5,
+    no_experience: 6,
+    experience1: 7,
+    experience2: 8,
+    experience3: 9,
+    experience4: 10,
+    experience5: 11
+  }
```

Comment:
One feature I'm interested to explore for a **later** iteration, would be the notion of a custom segment which could apply to an arbitrary set of users.

An example segment could be something like "CodeLand participants" which could be "refreshed" via admin API — of course, with the trade-offs that it doesn't correspond to a native query and needs to be managed custom.

What do you think about that? I love the idea of this base automated segmentation which will automatically stay up-to-date, work out of the box, and generally be more "powerful" in a lot of ways.

I bring this up now as a design consideration, not necessarily as a near-term implementation, but what do you think about this as part of the story?